### ISSUE TITLE
Specify parserOptions on the command line

### ISSUE THREAD

**Issue by ljharb** [eslint/eslint#5495]
**What version of ESLint are you using?**
v2.3 - but this applies to all of them.

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**
Default.

**What did you do? Please include the actual source code causing the issue.**
Attempted to provide `--parserOptions` on the command line.

**What did you expect to happen?**
To be able to specify `--parserOptions` on the command line.

**What actually happened? Please include the actual, raw output from ESLint.**
`Invalid option '--parserOptions' - perhaps you meant '--parser'?`

(Per https://github.com/eslint/eslint/issues/5200#issuecomment-193101659)

**Comment by mysticatea** [comment]
:+1: though I prefer `--parser-options`.

**Comment by ljharb** [comment]
sure, no bikeshed from me on what it's called - i just want to be able to do on the command line anything i can do in an eslintrc.

**Comment by nzakas** [comment]
What would the format of this option be?

**Comment by platinumazure** [comment]
Since we use optionator for levn/options, I think the simplest approach would be as an object (in levn dialect), so you could throw in ecmaVersion and ecmaFeatures.

Works pretty well for the scalars in parserOptions:

```bash
eslint --parser-options=ecmaVersion:7,sourceType:module
```

Not sure how well levn handles the nested object case (ecmaFeatures) and what shell escapes are needed to get braces working, but that would look something like this:

```bash
eslint --parser-options=ecmaFeatures:{jsx:true},ecmaVersion:6
```

...Or something like that. Anyway, it's just one idea, hopefully others will chip in.

**Comment by mysticatea** [comment]
I think it's the same format as `--rule` option.

**Comment by nzakas** [comment]
OK. @ljharb do you want to submit a PR for this?

**Comment by ljharb** [comment]
I'm not sure I understand `eslint` internals well enough to make a PR.

**Comment by nzakas** [comment]
This is pretty easy because parserOptions already exists. 

First, add the CLI option here:
https://github.com/eslint/eslint/blob/master/lib/options.js

Then here is where it gets translated from the CLI to be passed to CLIEngine:
https://github.com/eslint/eslint/blob/master/lib/cli.js#L53

Then, add a line here to store it:
https://github.com/eslint/eslint/blob/master/lib/config.js#L168

And merge into the config around here in a similar manner:
https://github.com/eslint/eslint/blob/master/lib/config.js#L253

Then, just need to update the related docs.

Pretty sure that will do it. It's not complicated, just a bunch of small steps.

**Comment by ljharb** [comment]
Thanks, that's a lot of guidance! I'll see what I can do.
